### ISSUE TITLE
Multiple `stratum_default_ID' lines in `searchParams.py` output

### ISSUE THREAD

**Issue by gothub** [RHESSys/ParamDB#1]
@jschoate discovered that `searchParams.py` is outputting multiple `stratum_default_ID` lines, when only one is expected. For example, the following commands reproduce the problem:

```
mv params.sqlite params.sqlite.bak
./createParamDB.sh 
./searchParams.py -v --searchType=hierarchical --name="Evergreen" --format=param 
... [output lines deleted for brevity]
0.000240 specific_snow_capacity 
4 stratum_default_ID 
1 stratum_default_ID 
-999.900000 ustar_overu 
... [output lines deleted]
```
In this case, only `1 stratum_default_ID` should be output, if I'm remembering correctly (after 5 years of not working on this project)

I haven't found the problem yet after reviewing the source and some testing. Possibly @selimnairb has some insight into the problem, as he has working on this program as well.

**Comment by gothub** [comment]
@jschoate it appears that parameters with names such as `*_default_ID` are not supposed to be in the database, but there are actually about 30. When a search is performed, the `default_ID` for the requested class is supposed to be calculated:
https://github.com/RHESSys/ParamDB/blob/master/rhessys/params.py#L676

So the search shows both the calculated and the retrieved (from the db) values. I'm not sure how these values got into the database, as they are supposed to be filtered from the input .def files:

https://github.com/RHESSys/ParamDB/blob/master/rhessys/params.py#L568

Either this section of code is not working correctly, or the `default_ID` parameters got into the database some other way. Maybe you could remind me of the proper handling of these type of parameters, then we could come up with a course of action to fix this.

**Comment by jschoate** [comment]
@gothub Each parameter definition file includes a *_default_ID, so when a definition file is read in the parameter *_default_ID is read in with all the other parameters. I tried reading in a stratum def file without the stratum_default_ID, but it errors and won't read the def file in, so it is apparently expected by the paramDB. The *_default_ID is a required parameter in a definition file, as it links with the ID in the worldfile, so RHESSys knows what characteristics to apply to a given area. The paramDB should I suppose either ignore the *_default_ID when it is read in, or overwrite it with the ID calculated by the paramDB upon output.

**Comment by gothub** [comment]
@jschoate - I'm not remembering why paramDB would calculate the *_default_ID values, instead of just using the ones read in from a parameter file. Right now it is using the 'default_id' field from the `class` table, as the value for *_default_ID for the requested class for a search, e.g. `evergreen` as the class.

**Comment by jschoate** [comment]
@gothub - To my recollection, the paramDB was going to assign an ID to base classes, i.e. evergreen 1, deciduous 2, grass 3, and nonveg 4. Then more specific types in a base class would build on the base class number, i.e. eucalyptus is a type of evergreen, so it would be assigned an ID of 11 (the first number representing the base class, the second number a specific type of the base class), and so on. The paramDB does appear to be doing this - however, it is also keeping the original stratum_default_ID that was read in from the def file.
Additionally, for the soils - the paramDB is outputting the original patch_default_ID read in from the .def file, but is outputting an additional soil_default_ID parameter. It should only output the parameter patch_default_ID and no parameter called soil_default_ID. And in this case - both of those parameters seem to be using the original patch_default_ID value read in from the .def file originally, and the paramDB does not appear to be assigning a number based on classification scheme.

**Comment by gothub** [comment]
@jschoate I found the spot in the python code where it is supposed to be filtering out the `*_default_ID` params but isn't. I have a fix, but will need to be added as a contributor to this repo. If you prefer, I can use a pull request instead.

**Comment by jschoate** [comment]
@gothub - I have added you as a collaborator with permissions to push to the code. You should have received an invite from github. Thank you!

**Comment by gothub** [comment]
The `readParamFile` function (in rhessys/paramss.py) now properly filters out `*_default_ID` parameters from .def files that are being read into the database. Parameter with the name `patch_default_ID` are not filtered however, which I believe is the desired behavior, so for these parameters, the value read in from the .def file is stored in the db, and is returned during a search.

This bug was fixed in commit 3202447575a2a1d5c86880cd615daadac45504c8

**Comment by jschoate** [comment]
@gothub - Stratum is working properly now - thank you!
